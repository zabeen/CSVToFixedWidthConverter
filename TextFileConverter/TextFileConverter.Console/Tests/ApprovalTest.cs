using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using System;
using System.IO;
using System.Text;
using TextFileConverter.Console;

namespace TextFileConverter.Console.Tests
{
    [TestFixture]
    [UseReporter(typeof(NUnitReporter))]
    public class ApprovalTest
    {
        [Test]
        public void VerifyFixedWidthOutput()
        {
            var converter = new Converter($"{TestContext.CurrentContext.TestDirectory}\\Files\\input.csv");
            converter.ConvertCsvToFixedWidth($"{TestContext.CurrentContext.TestDirectory}\\Files\\output.txt");
            Approvals.Verify(converter.Output);
        }
    }
}
