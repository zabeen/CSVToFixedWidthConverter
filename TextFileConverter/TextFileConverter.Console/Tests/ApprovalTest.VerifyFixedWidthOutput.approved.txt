| Pub Date    |                       Title | Authors                         |
|=============================================================================|
| 28 Nov 2008 |             Learning C# 3.0 | Jesse Liberty & Brian MacDonald |
| 16 Sep 2013 |               Head First C# | Jennifer Greene & Andrew Ste... |
| 27 Oct 2015 | Learn C# in One Day and ... | Jamie Chan                      |